parsing/parchmint-parser.js

```diff
@@ -204,6 +220,60 @@ class ParchmintParser {
                 compObj['entity'], ports);
     }
 
+    /**
+     * Parse a JSON object for the Connections.
+     *
+     * Fills the connections map with Connections where the keys are Layer IDs,
+     * and the values are arrays of Connections that exist on that Layer. Sets
+     * the parser invalid if a duplicate ID is found.
+     *
+     * @since 1.0.0
+     *
+     * @param {object} jsonObj  A parsed JSON object representing the Parchmint
+     *                          file.
+     */
+    parseConnections(jsonObj) {
+        jsonObj['connections'].forEach((connValue, index) => {
+            if (!this.isUniqueID(connValue['id'])) {
+                this.valid = false;
+                console.log('Parser: Duplicate ID (' + connValue['id'] + ') found in "connections" key. Skipping' +
+                        ' Connection with name ' + connValue['name'] + ' at index ' + index + '.');
+            } else {
+                let tempConn = this.getParsedConnection(connValue);
+
+                if (this.connections.has(connValue['layer'])) {
+                    this.connections.get(connValue['layer']).push(tempConn);
+                } else {
+                    this.connections.set(connValue['layer'], [tempConn]);
+                }
```

Comment:
This section of functionality seems to be quite prevalent. Is there a structure that can do this type of check for us, would it be worth building one, or could we write a function to do it for us?

Reply:
There is now a function ```addToMap``` that does this.